### DISCUSSION TITLE
Do you prefer Checkboxes or Switches?

### DISCUSSION THREAD

**Discussion by naveensingh** [FossifyOrg/General-Discussion#78] [Polls]
Currently, all apps use checkboxes as a toggle button and I want to replace them with switch preferences.

**Screenshots:**
| Checkbox (current) | Material Switch with check mark ✔ | Material Switch |
| --- | --- | --- |
| ![1](https://github.com/FossifyOrg/General-Discussion/assets/36371707/b8875589-2acd-4e27-94f7-4b45d1e73aa0) | ![2](https://github.com/FossifyOrg/General-Discussion/assets/36371707/f4dbbab5-f618-4f67-8ef2-145201dd8d2d) | ![3](https://github.com/FossifyOrg/General-Discussion/assets/36371707/ede2df93-3df0-490f-aef3-f38521fc6e87) |

**Comment by ghost** [comment]
It is clear that checkboxes are a little outdated.

However, I’m not convinced by the material switches with check marks, it overloads them a little and it makes me think of an accessibility option (I know that there is a similar option on iPhone but with I/O instead of the check mark) so I don't really like the idea of generalising them in the whole app suite, even if I admit they look pretty good.

**Comment by ghost** [comment]
Maybe I'm old fashioned, but I'm fine with the "Outdated" checkbox, it's simple. Though to be honest, I'd be fine with any of them even if the checkbox is my preference.

**Comment by inson1** [comment]
My preference is Material Switch 

Plus its used almost in every app

**Comment by Mr-Bajs** [comment]
Checkboxes are mainly supposed to be used when you can choose several options. But sence these are binary settings that are either on or off it's wise to change them to switches. 

The Material Switches looks the best and integrates the best with android in my opinion. It would improve the visual style of the apps.

**Comment by LucaMkIII** [comment]
I like the checkboxes because they're clean and simple and *smaller*, so they don't clutter up my screen. However, I doubt that my preference (de-cluttered) is typical, and I don't mind material switches. (Not a fan of material switches with check marks, personally, but I'm also not colorblind.)

**Reply by inson1** [reply]
"I'm also not colorblind" that is good point to make

But they are just not so pretty like normal switches

**Reply by Cwpute** [reply]
I am slightly colwurblind, and vhile i agree it could be a problem for some, you have look at the details: when off, the slider has a contour line and its button is slightly smaller, and is on the left (the direction most associated with regression, as opposed to right, associated with progression).
Also i'm pretty sure the contrast is different enough, which isn't affected by colourblindness: when the switch is off it is quite dim, greyish, and when it's on, the slider has a bright hue and its button is a very bright white.

**Reply by LucaMkIII** [reply]
Thank you so much for commenting! I mentioned that I'm not colorblind, and it goes both ways - things that may affect colorblind users may not affect me, but I also can't speak for what does/does not affect colorblind users. This is a good perspective that (I hope) helps me understand this situation better.

I still prefer checkboxes... But that's my opinion, and I'm still OK with whatever.

**Reply by naveensingh** [reply]
In the future, I think we'll add an accessibility sub-setting in all apps. The accessibility settings will allow people to for example switch between the default material switch and switches with checkmarks, another example is [this issue](https://github.com/FossifyOrg/Calendar/issues/103).

**Reply by rsb-23** [reply]
Brain focuses on 1 thing and ignores other, based on their habit. 
So, I choose `toggle with checkmark` as default. It has both - which makes it intuitive for all.

**Comment by Wintandre** [comment]
Overall, I don't really have a strong opinion towards any of them. I selected the switch with checkbox only because the added check mark removes any ambiguity as to whether it's on or not, regardless of the colour used or size of the switch's circle in the on/off states.

**Comment by seankhl** [comment]
If the philosophy is to be as uniform with vanilla android as possible, material switch is the clear choice. I would argue that this philosophy best aligns with the "simple" philosophy of the apps. We don't want our apps to do some custom thing, but rather the most simple, standardized thing, and that means conforming to MD3 where sensible. A strong vote for material switches from me.

**Reply by tom93** [reply]
To add a reference, the following is from the [Android settings design guidelines](https://source.android.com/docs/core/settings/settings-guidelines#checkbox):

> #### Checkbox
>
> Use switches over checkboxes when possible.
> 
> Checkboxes can be used:
> 
>  * For negative actions like restricting apps or blocking a service.
>  * To avoid having too many switches on the screen.

**Comment by GitCurious** [comment]
**Switches** are a little bigger and therefore easier for _me_ to click on more quickly, old eyes/small screen etc

**Reply by tom93** [reply]
FYI you can normally tap the label instead of having to aim for the small checkbox.

**Comment by Danny3** [comment]
I'm kinda divided on this!
I prefer checkboxes as it's more clear to me if something is enabled or not.
With toggles I'm normally confused if  pushed towards the left or right side means disabled or enabled.
But I guess that's why colors and even tick marks were added so this mitigates the problem.
Then if  it integrates better the rest of the system that's a plus, even though a very small one.
The fact that switches are bigger and easier to see and touch than checkboxes is another plus.
So from me it's 1 for checkboxes and 2 for switches.

But if you're looking for perfection and making me and probably other people too happy, then add a settings to choose which kind of toggles we want.
If you can make it for all apps would be even better and I don't think I will prefer different things depending on the app.

**Reply by LucaMkIII** [reply]
A toggle to toggle toggles! (I just wanted to say that.)

The idea has merit, especially if the poll is split between two options.

But one thing I like about these apps is that they're simple. The simpler they are, the less likely they are to have strange bugs pop up. When I use these apps, I give up some (nice-to-have) flexibility in return for glitch-free features that matter to me more.

*Edit: Clearer wording.*

**Comment by george9eg** [comment]
I'm not colorblind, but I'm using an Onyx BOOX Note Air 3C, which, while it supports color, the colors are a bit wonky and many of them are not distinct. Perhaps I'm slightly dyslexic or something, because I have to _think_ about which way is which, meaning that too much cognitive effort is involved. I find that the material switch graphics that are lighter / dimmer when in the "off" position confuse me--dimming / greying-out / stippling a control still means to me that it is _disabled_, not off, meaning that it is not clickable because the choice is not applicable in the current state.

So I like the idea of making the graphics much more visually distinct, as adding the checkmark does. I'm not particularly fond of the use of the checkmark as it seems to mix visual metaphors. However, I don't think the check mark is ambiguous or confusing.

**Comment by Secret-chest** [comment]
Checkboxes should be used when at least one of these is true:
* in a form (not instant);
* a list of similar options;
* subordinate to another control.

**Comment by DeMiro5001** [comment]
Everyone has his own preferences. Can this be changed in a "display settings" section in settings, where user can choose from a dropdown list? I don't know if it's doable or complicated.
Anyway, thanks a lot for your work

**Comment by naveensingh** [comment]
Fossify Gallery 1.2.0 is out with material switches in app settings. 

Since plenty of people still voted for 'Switches with checkmarks', I have added it as an accessibility option to the yet-to-be-released Thank You app. Enabling `Show checkmarks on switches` in the Thank You app will enable checkmarks in all Fossify apps.

**Comment by antodc** [comment]
Just to let you notice that Fossify Phone 1.3.0 still has checkboxes in ‘Manage shown tabs’. Same thing for Fossify Contacts 1.1.0, which has still checkboxes also in ‘Manage shown contact fields’.

**Reply by naveensingh** [reply]
That is expected. Checkboxes are the right component to use when you have multiple options to choose from.

**Reply by antodc** [reply]
Then, why are there switches in ‘Manage blocked numbers’?

**Reply by naveensingh** [reply]
> when you have multiple options to choose from.

I meant multiple options as in the 'Manage shown tabs' or 'Manage shown contact fields' dialog where you need to select one or more options from a list. Switches are best for enabling/disabling independent features i.e. changes take effect immediately.

Guidelines:
 - https://m3.material.io/components/switch/guidelines
 - https://m3.material.io/components/checkbox/guidelines

**Comment by naveensingh** [comment]
It was finally done in all apps.
